corehq/apps/zapier/signals/receivers.py

```diff
@@ -1,26 +1,46 @@
 from django.db.models.signals import pre_save, post_delete
 from django.dispatch import receiver
 
-from corehq.apps.repeaters.models import FormRepeater
+from corehq.apps.repeaters.models import FormRepeater, CaseRepeater
 from corehq.apps.zapier.models import ZapierSubscription
 
 
 @receiver(pre_save, sender=ZapierSubscription)
 def zapier_subscription_pre_save(sender, instance, *args, **kwargs):
+    """
+    Creates a repeater object corresponding to the type of trigger (form or case)
+    """
     if instance.pk:
         return
-    repeater = FormRepeater(
-        domain=instance.domain,
-        url=instance.url,
-        format='form_json',
-        include_app_id_param=False,
-        white_listed_form_xmlns=[instance.form_xmlns]
-    )
+
+    if instance.event_name == "new_form":
+        repeater = FormRepeater(
+            domain=instance.domain,
+            url=instance.url,
+            format='form_json',
+            include_app_id_param=False,
+            white_listed_form_xmlns=[instance.form_xmlns]
+        )
+
+    elif instance.event_name == "new_case":
+        repeater = CaseRepeater(
+            domain=instance.domain,
+            url=instance.url,
+            format='case_json',
+            white_listed_case_types=[instance.case_type],
+        )
+
     repeater.save()
```

Comment:
Could you could get into a situation where neither of these conditions will be true?  This will fail hard in that case. It might be nice to handle that gracefully if you think that's possible.